### PR TITLE
vim-patch:b529cfbd04c0

### PR DESCRIPTION
**Pull request by clason**
Update runtime files
https://github.com/vim/vim/commit/b529cfbd04c02e31cfa88f2c8d88b5ff532d4f7d

### CODE REVIEW

**Comment by clason** [comment]
@gpanders

**Comment by gpanders** [comment]
🙌

**Review by seandewar — COMMENTED** [review]
Including the [latest `shellmenu.vim`](https://github.com/vim/vim/blob/b529cfbd04c02e31cfa88f2c8d88b5ff532d4f7d/runtime/pack/dist/opt/shellmenu/plugin/shellmenu.vim) also looks safe (though the comment advising not to set `<` in `cpo` is moot)

**Comment by clason** [comment]
> Including the [latest `shellmenu.vim`](https://github.com/vim/vim/blob/b529cfbd04c02e31cfa88f2c8d88b5ff532d4f7d/runtime/pack/dist/opt/shellmenu/plugin/shellmenu.vim) also looks safe (though the comment advising not to set `<` in `cpo` is moot)

Really? It is a complete replacement...

File: runtime/doc/map.txt [hunk 1]
```diff
@@ -610,19 +610,22 @@ two bytes 0xc3 0xa1.  You don't want the 0xc3 byte to be mapped then or
...
+	map <Leader>A  oanother line<Esc>
 Works like: >
-	:map \A  oanother line<Esc>
-But after: >
-	:let mapleader = ","
+	map \A  oanother line<Esc>
+But after (legacy script): >
+	let mapleader = ","
+Or (Vim9 script): >
+	g:mapleader = ","
```
[the remark this answers is not in the document]
Comment (clason) [reply runtime/doc/map.txt:623]: 😱 

Nobody expects the Vim9 script!

**Comment by seandewar** [comment]
Looks like an improvement to me. It organizes the menus under one `ShellMenu` sub-menu and uses key notations rather than literal keycodes.

File: runtime/doc/undo.txt [hunk 2]
```diff
@@ -108,13 +108,13 @@ change again.  But you can do something like this: >
 
 After this a "u" command will undo the delete command and the previous
 change.
-							*undo-break*
-To do the opposite, break a change into two undo blocks, in Insert mode use
-CTRL-G u.  This is useful if you want an insert command to be undoable in
+						*undo-break* *undo-close-block*
```
Comment (zeertzjq) [inline runtime/doc/undo.txt:111]:
```suggestion
					*undo-break* *undo-close-block*
```

File: runtime/synmenu.vim [hunk 3]
```diff
@@ -18,6 +18,10 @@ fun! SetSyn(name)
   else
     let name = a:name
   endif
+  if name == "whitespace"
+    " do not replace the filetype but add whitespace on top
+    filetype = &ft .. ".whitespace"
```
Comment (zeertzjq) [inline runtime/synmenu.vim:23]:
```suggestion
    let name = &ft .. ".whitespace"
```

File: runtime/synmenu.vim [hunk 4]
```diff
@@ -18,6 +18,10 @@ fun! SetSyn(name)
   else
     let name = a:name
   endif
+  if name == "whitespace"
```
Comment (zeertzjq) [inline runtime/synmenu.vim:21]: Although it doesn't affect behavior, I guess using `a:name` here matches the code better?
```suggestion
  if a:name == "whitespace"
```

**Review by zeertzjq — APPROVED** [review]

**Review by seandewar — APPROVED** [review]

File: runtime/doc/options.txt [hunk 5]
```diff
@@ -3571,7 +3571,7 @@ A jump table for the options with a short description can be found at |Q_op|.
 	help.  (Note that previously setting the global option to the empty
 	value did this, which is now deprecated.)
 	When the first character is ":", the command is invoked as a Vim
-	command prefixed with [count].
+	Ex command with [count] added as an argument if it is not zero.
```
Comment (zeertzjq) [inline runtime/doc/options.txt:3574]: I just remembered Nvim does not support this.

**Pull request merged**